ui/src/components/table/tests/CustomTableRow.test.tsx

```diff
@@ -0,0 +1,112 @@
+    const props = {
+        page: 'inputs',
+        serviceName: 'example_input_one',
+        handleRequestModalOpen,
+        handleOpenPageStyleDialog,
+        displayActionBtnAllRows: false,
+    } satisfies ITableWrapperProps;
+
+    server.use(
+        http.get('/servicesNS/nobody/-/splunk_ta_uccexample_example_input_one', () =>
+            HttpResponse.json(MockRowData)
+        ),
+        http.post('/servicesNS/nobody/-/splunk_ta_uccexample_example_input_one/aaaaaa', () =>
+            HttpResponse.json(MockRowDataTogglingResponseDisableTrue)
+        )
+    );
+
+    setUnifiedConfig(SIMPLE_NAME_TABLE_MOCK_DATA_WITH_STATUS_TOGGLE_CONFIRMATION);
+
+    render(
+        <TableContextProvider>
+            <TableWrapper {...props} />
+        </TableContextProvider>,
+        { wrapper: BrowserRouter }
+    );
+});
+
+it('Status toggling with acceptance model displayed correctly', async () => {
+    const inputSwitches = await screen.findAllByRole('switch');
+
+    // open accept modal for first switch
+    await userEvent.click(inputSwitches[0]);
+
+    const acceptModal = await screen.findByRole('dialog');
+
+    expect(acceptModal).toBeInTheDocument();
+
+    const headerText = screen.getByText('Make input Inactive?');
+    expect(headerText).toBeInTheDocument();
+
+    const warningMessage = screen.getByText('Do you want to make input Inactive?');
+    expect(warningMessage).toBeInTheDocument();
+
+    const noBtn = screen.getByRole('button', { name: 'No' });
+    expect(noBtn).toBeInTheDocument();
+
+    const yesBtn = await screen.findByRole('button', { name: 'Yes' });
+    expect(yesBtn).toBeInTheDocument();
+
+    await userEvent.click(noBtn);
+
+    expect(acceptModal).not.toBeInTheDocument();
+});
+
+it('Status toggling with acceptance model toggles state', async () => {
+    const inputSwitches = await screen.findAllByRole('switch');
+
+    // open accept modal for first switch
+    await userEvent.click(inputSwitches[0]);
+
+    const acceptModal = await screen.findByRole('dialog');
+
+    expect(acceptModal).toBeInTheDocument();
+
+    const yesBtn = await screen.findByRole('button', { name: 'Yes' });
+    expect(yesBtn).toBeInTheDocument();
+
+    await userEvent.click(yesBtn);
+
+    const inactiveStatusButton = await screen.findByText('Inactive', {
+        selector: '[data-test="status"]',
+    });
+
+    expect(inactiveStatusButton).toBeInTheDocument();
+
+    server.use(
+        http.post('/servicesNS/nobody/-/splunk_ta_uccexample_example_input_one/aaaaaa', () =>
+            HttpResponse.json(MockRowDataTogglingResponseDisableFalse)
+        )
+    );
+
+    await userEvent.click(inputSwitches[0]);
+    const yesBtn2 = await screen.findByRole('button', { name: 'Yes' });
+    await userEvent.click(yesBtn2);
+
+    const inactiveStatusButtonAfterChange = screen.queryByText('Inactive', {
+        selector: '[data-test="status"]',
+    });
+
+    expect(inactiveStatusButtonAfterChange).toBeNull();
+});
```

Comment:
it would be nice if you extract setup and add testing negative scenario - when modal is closed without confirmation